src/afl-forkserver.c

```diff
@@ -292,19 +293,31 @@ void afl_frk_srv_start(afl_forkserver_t *frk_srv, char **argv) {
 
   if (frk_srv->exec_tmout) {
 
-    it.it_value.tv_sec = ((frk_srv->exec_tmout * FORK_WAIT_MULT) / 1000);
-    it.it_value.tv_usec = ((frk_srv->exec_tmout * FORK_WAIT_MULT) % 1000) * 1000;
+    fd_set readfds;
 
-  }
+    FD_ZERO(&readfds);
+    FD_SET(frk_srv->fsrv_st_fd, &readfds);
+    timeout.tv_sec = ((frk_srv->exec_tmout * FORK_WAIT_MULT) / 1000);
+    timeout.tv_usec = ((frk_srv->exec_tmout * FORK_WAIT_MULT) % 1000) * 1000;
+
+    int sret = select(frk_srv->fsrv_st_fd + 1, &readfds, NULL, NULL, &timeout);
 
-  setitimer(ITIMER_REAL, &it, NULL);
+    if (sret == 0) {
 
-  rlen = read(frk_srv->fsrv_st_fd, &status, 4);
+      if (frk_srv->child_pid > 0) {
 
-  it.it_value.tv_sec = 0;
-  it.it_value.tv_usec = 0;
+        frk_srv->child_timed_out = 1;
+        kill(frk_srv->child_pid, SIGKILL);
 
-  setitimer(ITIMER_REAL, &it, NULL);
+      }
+
+    } else {
+
+      rlen = read(frk_srv->fsrv_st_fd, &status, 4);
+
+    }
+
+  }
```

Comment:
The case `exec_tmout == 0` no longer reads 4 bytes and will fail

Reply:
I think exec tmout cannot be zero. Because even if we dont pass it the `-t` option, it still takes a certain default timeout.

Reply:
What about `-t 0`?

Reply:
Doesn't work. It gives a timeout too low warning and exits  :sweat_smile: